server/service/integration_mdm_test.go

```diff
@@ -4764,6 +4764,10 @@ func (s *integrationMDMTestSuite) TestSSO() {
 	s.runWorker()
 	require.Contains(t, lastSubmittedProfile.URL, "https://example.com/api/mdm/apple/enroll?token=")
 	require.Equal(t, "https://example.com/mdm/sso", lastSubmittedProfile.ConfigurationWebURL)
+
+	// hitting the callback with an invalid session id results in a 4xx
+	rawSSOResp := base64.StdEncoding.EncodeToString([]byte(`<samlp:Response ID="_7822b394622740aa92878ca6c7d1a28c53e80ec5ef"></samlp:Response>`))
+	s.DoRawNoAuth("POST", "/api/v1/fleet/mdm/sso/callback?SAMLResponse="+url.QueryEscape(rawSSOResp), nil, http.StatusNotFound)
```

Comment:
I wonder if a 404 could not be misleading, e.g. the user would look at their configuration to check if the callback path is properly configured/points to the right path vs returning a 401 unauthorized, as this is basically a failed attempt at SAML authorization IIUC?

This is definitely a minor detail though, feel free to marge as-is if it is urgent to get the fix in the release (or if you disagree and feel 404 is still a better option).

Reply:
Agree `401` is more intuitive for the user.